library/views.py

```diff
@@ -33,3 +34,38 @@ def post(self,request):
             return JsonResponse({'book_save':'SUCCESS'}, status=200)
         except KeyError:
             return JsonResponse({'message':'INVAILD_KEYS'}, status=400)
+
+
+class LibraryBookListView(View):
+    @check_auth_decorator
+    def get(self, request, user_id):
+        user_id  = request.user
+        ordering = request.GET.get('ordering', '1')
+
+        if int(ordering) == 1:
+            books = LibraryBook.objects.select_related(
+                'book', 'library').filter(library__user_id=user_id).order_by('-created_at')
+
+        elif int(ordering) == 2:
+            books = LibraryBook.objects.select_related(
+                'book','library').filter(
+                    library__user_id = user_id).order_by('book__title')
+
+        elif int(ordering) == 3:
+            books = LibraryBook.objects.select_related(
+                'book','library').filter(
+                    library__user_id = user_id).order_by('book__author')
+
+        elif int(ordering) == 4:
+            books = LibraryBook.objects.select_related(
+                'book', 'library').filter(
+                    library__user_id = user_id).order_by('-book__publication_date')
+
+        book_list = {
+            "libraryBook" : [{
+                "id"     : library.book.id,
+                "title"  : library.book.title,
+                "image"  : library.book.image_url,
+                "author" : library.book.author
+            } for library in books]}
```

Comment:
```suggestion
    def get(self, request, user_id):
        user_id  = request.user
        ordering = request.GET.get('ordering', '1')

        conditions = {
            1 : '-created_at',
            2 : 'book__title',
            3 : 'book_author',
            4 : '-book_publication_date'
        }
        
        books = LibraryBook.objects.select_related('book', 'library').filter(library__user_id = user_id).order_by(conditions[int(ordering)])

        book_list = {
            "libraryBook" : [{
                "id"     : library.book.id,
                "title"  : library.book.title,
                "image"  : library.book.image_url,
                "author" : library.book.author
            } for library in books]}
```

수희님 현재 필터링 하는 부분의 코드가 if문 내부에 계속 반복되고 있습니다 :) order_by하는 조건만 달라지네요 :)
이 방법의 단점은, 조건이 하나 추가될 때마다 if문이 계속 추가되어야 한다는 점입니다. 아마 이건 수희님도 뭔가 바꾸고 싶으셨을 거라고 생각해요. 파이썬에서는 이런 경우에 딕셔너리를 자주 사용합니다 :)

코드가 어떤 의미인지는 파악하셨을 거라고 생각합니다. 혹시 이해가 안되신다면 찬찬히 한 번 생각해보시면 될 것 같습니다

Reply:
@soheon-lee 
네 어떤 의미인지 이해가 가네요 수정하겠습니다 
감사합니다!